core/src/test/java/org/apache/iceberg/util/TestLocationUtil.java

```diff
@@ -58,4 +58,30 @@ public void testStripTrailingSlashWithInvalidPath() {
           .hasMessage("path must not be null or empty");
     }
   }
+
+  @Test
+  void testStripTrailingSlashForRootPath() {
```

Comment:
```suggestion
  void testDoNotStripTrailingSlashForRootPath() {
```

this is because it shouldn't strip the trailing slash in this case